applications/GeoMechanicsApplication/custom_workflows/time_stepping.hpp

```diff
@@ -0,0 +1,77 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//                   Anne van de Graaf
+//
+
+#pragma once
+
+#include "processes/process.h"
+#include "time_step_end_state.hpp"
+#include "strategy_wrapper.hpp"
+#include "geo_mechanics_application_variables.h"
+
+#include <functional>
+#include <memory>
+
+
+namespace Kratos
+{
+
+class TimeStepExecutor
+{
+public:
+    using ProcessRef    = std::reference_wrapper<Process>;
+    using ProcessRefVec = std::vector<ProcessRef>;
+
+    void SetSolverStrategy(std::shared_ptr<StrategyWrapper> SolverStrategy)
+    {
+        mStrategyWrapper = std::move(SolverStrategy);
+    }
+
+    void SetProcessReferences(ProcessRefVec ProcessRefs)
+    {
+        mProcessRefs = std::move(ProcessRefs);
+    }
```

Comment:
In the end we changed a bit how these processes are passed. It will now be passed as a vector of std::weak_ptr. @WPK4FEM @avdg81 Do you think we should fix that as part of this PR, or in the timeloopexecutor?

Reply:
We will tie things together when implementing the time loop executor.